Extension/Services/TroopActionService.cs

```diff
@@ -78,25 +78,39 @@ public static (int, int) RecruitPrisoners(PartyVM partyVm, PartyScreenLogic part
                 }
             }
 
+            var playerSelectedCharacter = partyVm.CurrentCharacter.Character;
+
             int recruitedTypes = 0;
             int recruitedTotal = 0;
             List<Tuple<PartyCharacterVM, PartyScreenLogic.PartyCommand>> commands = new List<Tuple<PartyCharacterVM, PartyScreenLogic.PartyCommand>>();
-            while (recruitablePrisoners.Count > 0 && (partySpace > 0 || States.HotkeyControl))
+            foreach (var prisoners in recruitablePrisoners.OrderByDescending(vm => vm.Character.Tier))
             {
-                PartyCharacterVM prisoners = recruitablePrisoners.First();
+                if (recruitablePrisoners.Count == 0 || (partySpace == 0 && !States.HotkeyControl))
```

Comment:
Would like brace used or inline the `break;`

Reply:
Also no need to check `recruitablePrisoners.Count` if foreach is used

Reply:
OK and correct